src/MultiSearchPrompt.php

```diff
@@ -132,6 +134,20 @@ public function visible(): array
         return array_slice($this->matches(), $this->firstVisible, $this->scroll, preserve_keys: true);
     }
 
+    /**
+     * Toggle all options.
+     */
+    protected function toggleAll(): void
+    {
+        if (count($this->values) === count($this->matches)) {
+            $this->values = [];
+        } else {
+            $this->values = $this->isList()
+                ? array_combine(array_values($this->matches), array_values($this->matches))
+                : array_combine(array_keys($this->matches), array_values($this->matches));
+        }
+    }
```

Comment:
I'm not totally sure about this behaviour.

When I was playing around with it, I tried the following:

1. Perform a search (e.g. search for "james" in the playground) and select one or more options.
2. Perform another search that excludes the previously selected options (e.g. search for "nuno") and then use `Ctrl+A` to select all of the options.

I expected that the previously selected option(s) (i.e. James) would still be selected (which you can see by clearing the search or submitting it) but it wasn't.

I also tried this:

1. Perform a search (e.g. search for "james" in the playground) and select an option.
2. Perform another search that returns a single option that is not the previously selected option (e.g. search for "nuno") and then select that option using the space bar.
3. Now press `Ctrl+A`.

Due to the count comparison with the currently selected values, it will think nothing is selected and will try to select Nuno (who is already selected) and silently unselect James.

Curious to hear your thoughts on how you'd expect it to behave in these scenarios.

I think it should probably determine whether all of the current matches are included in the values to determine whether it's a select or unselect operation. And I think it should always maintain any values that were previously selected that aren't included in the current matches regardless of whether it's selecting-all or unselecting-all.

Reply:
> I think it should probably determine whether all of the current matches are included in the values to determine whether it's a select or unselect operation. And I think it should always maintain any values that were previously selected that aren't included in the current matches regardless of whether it's selecting-all or unselecting-all.

Yeah, I agree, it should probably be checking to see if all the current matches are included in the values, rather than checking the values/matches counts. 

I'll aim to get this PR updated today. 👍

Reply:
I've just pushed up a commit which should hopefully sort the behaviour. Let me know what you think.

Reply:
Nice! The new behaviour feels really intuitive.